tests/test_integration.py

```diff
@@ -1,14 +1,13 @@
 import os
 import tempfile
 import shutil
-
 from unittest import TestCase
 
 from ffmpeg_tools import codecs
 from ffmpeg_tools import commands
 from ffmpeg_tools import formats
 from ffmpeg_tools import meta
-from tests.utils import get_absolute_resource_path
+from tests.utils import get_absolute_resource_path, generate_sample_video
 
 
 class TestIntegration(TestCase):
```

Comment:
We need more integration tests and less tests that check if command is generated properly. Especially when this streams logic is so complicated.

Reply:
I agree to a degree. Please note that I actually put extra time into implementing a utility that would let me test it on actual videos (`generate_sample_video()`) because I don't even have enough sample files to do real integration tests for subtitles. All the other tests are more like unit tests. There's very little in terms of integration testing right now and if I just went with the flow that would not change a bit.

On the other hand, there will always be less integration tests than unit tests. It's just not feasible to test every corner case with integration tests because things can't be mocked and we need real videos for every test. It requires lots of of actual test data and makes the test suite very slow. We need both kinds of tests in right proportion.

I agree that currently there's too little of both types. In these recent branches I tried to change that by rigorously writing unit tests for everything I touched and also adding some integration tests where it made sense.

Also, the whole set of helpers for integration testing of ffmpeg results was left in the Golem repo when ffmpeg-tools was created. I didn't want to recreate all that here.

Reply:
I think you added great amount of tests, even more then I expected. But here I'm afraid only of this logic related to subtitles and replacing streams. We could either simplify it or make more tests.

Reply:
OK. I addressed simplifying the logic in a comment above (https://github.com/golemfactory/ffmpeg-tools/pull/26#discussion_r353795532).

But as a short term solution - if you have any particular test in mind I can add it or try to convert it to an integration test.